tests/integration/test_check_downloaded_output.py

```diff
@@ -0,0 +1,82 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import subprocess
+from os import path
+import tarfile
+from pathlib import Path
+
+import utils
+
+
+def test_check_downloaded_output(test_env):
```

Comment:
Would it be a good idea to reuse the completed request from `test_valid_data_in_request` to avoid needing to create a new request for each new test? I think this pattern would speed up the tests significantly over time. You could even use a pytest fixture that creates the request and then waits for it to complete. Once complete, it returns the request response. Then subsequent tests that use that fixture would just get the request response immediately.

I'll defer to your testing expertise on this though.

Reply:
I didn't realise we can do it like this. Some of the scenarios will still need to create the request on their own because of the different payload (e.g. dependency replacement) but the fixture will be used for the majority of tests.